tools/cldr-code/src/main/java/org/unicode/cldr/test/CoverageLevel2.java

```diff
@@ -179,38 +203,175 @@ public int getIntLevel(String path) {
+                    approvalRequirements.add(parts.toString());
+                } else if (parts.containsElement("coverageLevel")) {
+                    String match = parts.containsAttribute("match") ? coverageVariables.replace(parts.getAttributeValue(-1,
+                        "match")) : null;
+                    String valueStr = parts.getAttributeValue(-1, "value");
+                    // Ticket 7125: map the number to English. So switch from English to number for construction
+                    valueStr = Integer.toString(Level.get(valueStr).getLevel());
+
+                    String inLanguage = parts.containsAttribute("inLanguage") ? coverageVariables.replace(parts
+                        .getAttributeValue(-1, "inLanguage")) : null;
+                    String inScript = parts.containsAttribute("inScript") ? coverageVariables.replace(parts
+                        .getAttributeValue(-1, "inScript")) : null;
+                    String inTerritory = parts.containsAttribute("inTerritory") ? coverageVariables.replace(parts
+                        .getAttributeValue(-1, "inTerritory")) : null;
+                    Integer value = (valueStr != null) ? Integer.valueOf(valueStr) : Integer.valueOf("101");
+                    if (cldrVersion.getMajor() < 2) {
+                        value = 40;
+                    }
+                    CoverageLevelInfo ci = new CoverageLevelInfo(match, value, inLanguage, inScript, inTerritory);
+                    coverageLevels.add(ci);
+                } else if (parts.containsElement("coverageVariable")) {
+                    String key = parts.getAttributeValue(-1, "key");
+                    String value = parts.getAttributeValue(-1, "value");
+                    coverageVariables.add(key, value);
+                }
+            }
+            public void cleanup() {
+                CLDRConfig testInfo = ToolConfig.getToolInstance();
+                SupplementalDataInfo supplementalDataInfo2 = testInfo.getSupplementalDataInfo();
+                CoverageLevelInfo.fixEU(coverageLevels, supplementalDataInfo2);
+                coverageLevels = Collections.unmodifiableSortedSet(coverageLevels);
+            }
+        }
+
+        public RegexLookup<Level> makeCoverageLookup() {
+            RegexLookup<Level> lookup = new RegexLookup<>(RegexLookup.LookupType.STAR_PATTERN_LOOKUP);
+
+            Matcher variable = PatternCache.get("\\$\\{[A-Za-z][\\-A-Za-z]*\\}").matcher("");
+
+            for (CoverageLevelInfo ci : coverageLevels) {
+                String pattern = ci.match.replace('\'', '"')
+                    .replace("[@", "\\[@") // make sure that attributes are quoted
+                    .replace("(", "(?:") // make sure that there are no capturing groups (beyond what we generate
+                    .replace("(?:?!", "(?!"); // Allow negative lookahead
+                pattern = "^//ldml/" + pattern + "$"; // for now, force a complete match
```

Comment:
Are all patterns in coverage.xml really direct subpaths of /ldml/ ?

Reply:
Yes. We don't have provision for non-ldml coverage because we don't have any such items in ST.